app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,255 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+
+    private TaskService taskService;
+
+    private static final String TASK_TITLE = "test";
+    private static final String UPDATE_POSTFIX = "!!!";
+    private static final Long INVALID_ID = 0L;
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Nested
+        @DisplayName("만약 등록된 Task가 있다면")
+        class Context_has_task {
+            final int givenTaskCnt = 5;
+
+            @BeforeEach
+            void prepare() {
+                for (int i = 0; i < givenTaskCnt; i++) {
+                    taskService.createTask(getTask());
+                }
+            }
+
+            @Test
+            @DisplayName("Task 전체 리스트를 리턴한다.")
+            void it_return_tasks() {
+                assertThat(taskService.getTasks()).hasSize(givenTaskCnt);
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록된 Task가 없다면")
+        class Context_has_not_task {
+            @Test
+            @DisplayName("빈 리스트를 리턴한다.")
+            void it_return_tasks() {
+                assertThat(taskService.getTasks()).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록된 Task의 id 값이 주어진다면")
+        class Context_with_id {
+            Task givenTask;
+            Long givenId() {
+                return givenTask.getId();
+            }
+
+            @BeforeEach
+            void prepare() {
+                givenTask = taskService.createTask(getTask());
+            }
+
+            @Test
+            @DisplayName("등록된 task 정보를 리턴한다.")
+            void it_return_task() {
+                Task foundTask = taskService.getTask(givenId());
+
+                assertThat(foundTask).isNotNull();
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록되지 않은 Task의 id 값이 주어진다면")
+        class Context_with_invalid_id {
+            @Test
+            @DisplayName("TaskNotFoundException를 던진다")
```

Comment:
잘 작성하셨지만 예외 이름을 그대로 쓰는 것에 대해 주의할 점이 하나 있어요. 만약 예외 클래스 이름을 바꾸면 여기 테스트 설명도 같이 바꿔줘야겠죠? 그래서 저는 보통 이렇게 씁니다.
```suggestion
            @DisplayName("Task를 찾을 수 없다는 내용의 예외를 던진다")
```

Reply:
수정하겠습니다 !!